CONTRIBUTING.md

```diff
@@ -0,0 +1,115 @@
+
+The use of comments to note problems or future fixes is discouraged, and merges containing them will not be approved.
+Instead, create an issue to track the required change.
+
+<!--
+## Testing
+This section is to be drafted by whoever solves the issue relating to testing.
+
+-->
+
+## Branches And Commits
+
+The primary branch of this project is called 'mainline'.
+It represents the total summation of approved code changes.
+The tip of this branch will pass all linting and test stages.
+Tested stable versions are marked with git tags; there is no 'stable' branch.
+
+The mainline branch uses a 'semi-linear' branch history, with merge commits only being
+used to indicate the start and end of each applied topic branch.
+Rebasing code in topic branches is encouraged.
+
+Each commit in the repo should represent a logical, testable change.
+Ideally, they should also pass all linting and testing checks.
+When a branch is in active development, this does not need to be strictly adhered to,
+and you may wish to hold off on deciding on the commit order until the merge request.
+
+Commit messages, like code elements, starts with a summary of the intent of the change,
+and the full version can also include references to the issue(s) it solves and descriptions
+of the functional details of the changes.
+
+## Pull Requests: Getting Stuff Into The Project
+
+When a piece of work is ready for review (including feedback during development),
+you can open a pull request on GitHub. As with code and commits,
+the title and description of the request should describe intent and purpose first and foremost.
+
+If the changes are based on an issue, you can invite people involved in the issue as reviewers.
+GitHub also notifies the maintainers of all new pull requests.
+
+Pull requests are where this policy primarily gets enforced.
+The pull request template will outline the steps.
+The reviewers and maintainers will work with you to ensure that your code and commit history are correct and concise.
+
+## Reviewing Merge Requests: How to Give and Interpret Feedback
+
+Reviews on merge requests should be submitted using GitHub's review feature.
+Comments generally fall into four categories, but GitHub gives us no way to distinguish between them:
+ - Praise: "This is good."
+ - Question for the Author: "will this work if [condition]?"
+ - Alternative method: "this might be better as a [lanugage feature]."
+ - Actual problem: "Because of [x], this change won't work for cute cats."
+
+All four of these are suitable forms of feedback.
+This list is not exhaustive but forms a good framework for communicating what you expect the requester to do.
+When giving feedback of alternative methods or problems, include a clear description and example (pseudo)code or a link to documentation.
+
+It is up to the reviewer (or a maintainer) to mark a conversation as resolved, not the merge requester.
+An exception naturally applies to praise and other comments that do not need a response.
```

Comment:
Overall well written and clear.

Reply:
Praise: I do like this idea of marking your comments on what we should say.
Potential Optional Addition: do we want to mandate a list/mandate having these tone markers

Reply:
I don't think they _need_ tone markers, per se -- I'm hoping that just having the idea of thesallows for clearer communication (both in the form of people writing comments with language that makes it clearer, and also giving people a way to ask "is this a problem that has to be dealt with?" in a slightly more polite manner